### PR TITLE
script: split serialization tests for both `call_function` and `evaluate`

### PR DESCRIPTION
**Pull request by thiagowfx**
This is to try to fix some of the slow tests in the stability jobs.

- `test_result` -> `test_primitive_values` + `test_remote_values`
- `test_exception_details` -> `test_exception_details` + `test_exception_details_invalid` + `test_exception_details_await_promise`
- `test_argument` -> `test_argument` + `test_channel` + `test_remote_reference`

### CODE REVIEW

**Comment by thiagowfx** [comment]
Note: This is still a draft, but early comments would be appreciated.

I intend to do some more refactoring in the fixtures and to apply similar transformations to `call_function`.

File: webdriver/tests/bidi/script/call_function/invalid.py
Comment (thiagowfx) [inline webdriver/tests/bidi/script/call_function/invalid.py:408]: This function name was duplicated

**Review by jrandolf — APPROVED** [review]

File: webdriver/tests/bidi/script/call_function/strict_mode.py
Comment (whimboo) [inline webdriver/tests/bidi/script/call_function/strict_mode.py:28]:
```suggestion
        "value": 1,
```

Shouldn't this get a final comma as well? Or does only ESLint enforce it? This would apply to other changes as well.

File: webdriver/tests/bidi/script/evaluate/__init__.py
Comment (whimboo) [reply webdriver/tests/bidi/script/evaluate/__init__.py:149]: Well, you could just store the actual data as global constants and then have something like the following for the tests:

```
@pytest.mark.parametrize("expression, expected", PRIMITIVE_VALUES + REMOTE_VALUES
```

Means I would not store the result of `parameterize` but still use it as decorator which also makes it more obvious to use.

File: webdriver/tests/bidi/script/evaluate/exception_details_await_promise.py
Comment (whimboo) [inline webdriver/tests/bidi/script/evaluate/exception_details_await_promise.py:12]: Why not the following which we use in a lot of other places?

```suggestion
async def test_exception_details_await_promise(
    bidi_session, top_context, expression, expected
):
```

**Review by whimboo — CHANGES_REQUESTED** [review]
Just had a quick look and put some inline comments as suggestions.

**Comment by thiagowfx** [comment]
Comments have been addressed. Note that `call_function` still needs to be split.

I suggest to do whatever comes first:

- If I get to make the changes before review is complete, then let's submit everything together
- Otherwise if review is done first then I will send a follow up for call_function in another PR

File: webdriver/tests/bidi/script/evaluate/exception_details.py
Comment (OrKoN) [inline webdriver/tests/bidi/script/evaluate/exception_details.py:10]: this still looks like a lot of tests, I wonder if it would still fail sometimes.

**Comment by thiagowfx** [comment]
PR is done. Both call_function and evaluate were updated. No further changes are planned.

**Comment by thiagowfx** [comment]
PR done once again. @whimboo PTAL.

**Comment by juliandescottes** [comment]
> PR done once again. @whimboo PTAL.

For info, @whimboo is on PTO until July 31st, I can take a look instead (but we might need an admin merge to waive the flags from Henrik?)

**Comment by thiagowfx** [comment]
@juliandescottes ah, thanks for letting me know. Sure, would appreciate if you could take a look instead!

For admin merge I can ask one of the admins once it's approved.

**Comment by OrKoN** [comment]
@thiagowfx looks like it is still slow

**Review by OrKoN — APPROVED** [review]
LGTM (I suggest we land this and then split up the remaining test in a separate CL?)

**Comment by thiagowfx** [comment]
Since we have to wait for review anyway, I am splitting `arguments.py` in this PR still.

**Review by juliandescottes — APPROVED** [review]
Thanks, overall this looks fine. Did we consider using `# META: timeout=long`? Maybe we could avoid splitting exception_details if we used this?

**Comment by thiagowfx** [comment]
Splitting definitely helped, because `wpt-chrome-dev-stability` is now passing (which was the goal all along). What is the effect of `META: timeout=long`?

File: webdriver/tests/bidi/script/call_function/exception_details_await_promise.py
Comment (thiagowfx) [reply webdriver/tests/bidi/script/call_function/exception_details_await_promise.py:1]: Does it make sense to do both?

I am fine to add `META: timeout=long` if it helps in any way.

That said, observing the latest [CI results](https://github.com/web-platform-tests/wpt/pull/40887/checks?check_run_id=14922874639), splitting alone has helped.

File: webdriver/tests/bidi/script/call_function/exception_details_invalid.py
Comment (thiagowfx) [reply webdriver/tests/bidi/script/call_function/exception_details_invalid.py:33]: I'll just move them back to `exception_details` then, since it's only two tests.

**Review by OrKoN — APPROVED** [review]

**Comment by juliandescottes** [comment]
> Splitting definitely helped, because `wpt-chrome-dev-stability` is now passing (which was the goal all along). What is the effect of `META: timeout=long`?

This extends the individual timeout of a test from 25 seconds to 3 minutes (https://github.com/web-platform-tests/wpt/blob/d4b2f98bc582144047b002f7c6bc12f761ab95d8/tools/wptrunner/wptrunner/wpttest.py#L686-L687) 

It's quite a big difference, but when splitting doesn't feel natural or might create test files that will be hard to understand in the future, it's worth considering.

**Comment by thiagowfx** [comment]
> This extends the individual timeout of a test from 25 seconds to 3 minutes (

That's very good to know! Will consider it in the future.

In this case, given that we have undone some of the `invalid` splits, the end result is OK.

Arguably we could re-merge some of the `arguments.py` or `exception_details.py` tests, depending on preference. If you feel strong about this, let me know.
I'd still keep the `result.py` split either way.

File: webdriver/tests/bidi/script/call_function/exception_details_await_promise.py
Comment (thiagowfx) [reply webdriver/tests/bidi/script/call_function/exception_details_await_promise.py:1]: See comment below. Keeping it as is.

**Comment by thiagowfx** [comment]
@DanielRyanSmith ready to go

**Pull request merged**
